// @laser-dac/laserdock/dist/LaserdockLib.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendSamples = exports.setDacRate = exports.clearRingBuffer = exports.disableOutput = exports.enableOutput = exports.init = void 0;
const path = require("path");
const ffi = require("ffi-napi");
const ref = require("ref-napi");
const ArrayType = require('ref-array-di')(ref);
const Struct = require('ref-struct-di')(ref);
const LaserdockPoint = Struct({
    rg: 'uint16',
    b: 'uint16',
    x: 'uint16',
    y: 'uint16'
});
const LaserdockPointArray = ArrayType(LaserdockPoint);
// Windows 32-bit is not supported currently
const libPath = path
    .join(__dirname, '../sdk/liblaserdocklib')
    // Super super dirty hack to make this work with Electron; native dependencies
    // dont'get placed inside the "app.asar" bundle, but instead get placed in a separate directory called "app.asar.unpacked"
    .replace('app.asar', 'app.asar.unpacked');
const LaserdockLib = ffi.Library(libPath, {
    nodeInit: ['int', []],
    nodeEnableOutput: ['int', []],
    nodeDisableOutput: ['int', []],
    nodeSetDacRate: ['int', ['uint32']],
    nodeClearRingbuffer: ['int', []],
    nodeSendSamples: ['int', [LaserdockPointArray, 'uint32']]
});
function init() {
    return LaserdockLib.nodeInit();
}
exports.init = init;
function enableOutput() {
    return LaserdockLib.nodeEnableOutput();
}
exports.enableOutput = enableOutput;
function disableOutput() {
    return LaserdockLib.nodeDisableOutput();
}
exports.disableOutput = disableOutput;
function clearRingBuffer() {
    return LaserdockLib.nodeClearRingbuffer();
}
exports.clearRingBuffer = clearRingBuffer;
function setDacRate(rate) {
    return LaserdockLib.nodeSetDacRate(rate);
}
exports.setDacRate = setDacRate;
function sendSamples(points, numOfPoints) {
    return LaserdockLib.nodeSendSamples(points, numOfPoints);
}
exports.sendSamples = sendSamples;
